dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+                      "streaming-kafka-assembly/assembly"]
+
+    # Then build with default Hive version (0.13.1) because tests are based on
+    # this version
+    print "[info] Compile with Hive 0.13.1"
+    rm_r("lib_managed")
+    print "[info] Building Spark with these arguments:", 
+    print " ".join(hive_profile_args)
+
+    if AMPLAB_JENKINS_BUILD_TOOL == "maven":
+        exec_maven(hive_profile_args + base_mvn_args)
+    else:
+        exec_sbt(hive_profile_args + sbt_hive_goals)
+
+
+def detect_binary_inop_with_mima():
+    set_title_and_block("Detecting binary incompatibilities with MiMa",
+                        "BLOCK_MIMA")
+    run_cmd(["./dev/mima"])
+
+
+def run_scala_tests(test_suite=[]):
+    """Function to properly execute all tests pass in, as a list, from the
+    `determine_test_suite` function"""
+    set_title_and_block("Running Spark unit tests", "BLOCK_SPARK_UNIT_TESTS")
+
+    # ensure the test_suite is a set
+    if not isinstance(test_suite, set): 
+        test_suite = set(test_suite)
+
+    # if the Spark SQL tests are enabled, run the tests with the Hive profiles 
+    # enabled.
+    if "SQL" in test_suite:
+        sbt_maven_profile_args = \
+            os.environ.get(SBT_MAVEN_PROFILE_ARGS_ENV).split()
+        os.environ[SBT_MAVEN_PROFILE_ARGS_ENV] = \
+            " ".join(sbt_maven_profile_args + ["-Phive", "-Phive-thriftserver"])
+
+    # if we only have changes in SQL build a custom test string
+    if "SQL" in test_suite and "CORE" not in test_suite:
+        sbt_maven_test_args = ["catalyst/test",
+                               "sql/test",
+                               "hive/test", 
+                               "hive-thriftserver/test",
+                               "mllib/test"]
+    else:
+        sbt_maven_test_args = ["test"]
+
+    # get the latest sbt maven profile arguments
+    sbt_maven_profile_args = os.environ.get(SBT_MAVEN_PROFILE_ARGS_ENV).split()
+
+    print "[info] Running Spark tests with these arguments:",
+    print " ".join(sbt_maven_profile_args), 
+    print " ".join(sbt_maven_test_args)
+
+    if AMPLAB_JENKINS_BUILD_TOOL == "maven":
+        exec_maven(["test"] + sbt_maven_profile_args + ["--fail-at-end"])
+    else:
+        exec_sbt(sbt_maven_profile_args + sbt_maven_test_args)
+
+
+def run_python_tests(test_suite=[]):
+    set_title_and_block("Running PySpark tests", "BLOCK_PYSPARK_UNIT_TESTS")
+    
+    # Add path for Python3 in Jenkins if we're calling from a Jenkins machine
+    if AMPLAB_JENKINS:
+        os.environ["PATH"] = os.environ.get("PATH")+":/home/anaconda/envs/py3k/bin"
+
+    run_cmd(["./python/run-tests"])
+
+
+def run_sparkr_tests(test_suite=[]):
+    set_title_and_block("Running SparkR tests", "BLOCK_SPARKR_UNIT_TESTS")
+
+    if which("R"):
+        run_cmd(["./R/install-dev.sh"])
+        run_cmd(["./R/run-tests.sh"])
+    else:
+        print "Ignoring SparkR tests as R was not found in PATH"
+
+if __name__ == "__main__":
```

Comment:
Super minor nit, but do you mind moving the body of this `__main__` block into a method called `main()` and calling that `main()` method here?  That practice helps to avoid spurious variable shadowing warnings in linters like `pylint`.